### ISSUE TITLE
FoundryVTT: Whispered Rolls for Monsters and Players Show to Everyone

### ISSUE THREAD

**Issue by Koppartikus** [kakaroto/Beyond20#919]
**Describe the bug**
"Whisper Rolls" and "Whisper Monster Rolls" options are showing up as shown to everyone instead of having them hidden from players when rolls are being sent to Foundry VTT 9.242.

**NOTE:** While rolls do not work, using the "Beyond 20" button to show things like class features and monster traits (like multiattack) are sent whispered, and work as expected.

**To Reproduce**
1. Set "Whisper Rolls" to _Always Whisper_ and "Whisper Monster Rolls" to _Always Whisper Monster Rolls_
2. Click on any ability that requires a roll on a monster sheet or a player sheet (monster sheets can be rolled through encounters or on their own individual monster page)
3. Go to FoundryVTT and see that the rolls are shown to everyone instead of whispered to the DM

**Expected behavior**
Rolls should be hidden to players and shown as whispered to the DM to the DM.

**Screenshots**
Settings used for the test:
![Whisper Rolls](https://user-images.githubusercontent.com/25122550/150692420-b8b2097a-0f14-4dc8-8c5a-10b8ec4516ec.png)

Results in Foundry (note that rolls are not whispered, but features are):
![Whisper Rolls Foundry](https://user-images.githubusercontent.com/25122550/150692432-d284446a-e626-4ef2-9014-d9e79ffabd7d.png)

**Browser Info (please complete the following information):**
 - Used Firefox 96.0.2 (64-bit) and Chrome Version 97.0.4692.99 (Official Build) (64-bit) for testing - both exhibited the same results.

**Additional context**

- Latest Beyond20 version used for the test (2.6.0)

**Comment by kakaroto** [comment]
I just did a few tests, both with monsters and character sheets, and could not reproduce it, the whisper setting seems to be respected and all rolls (not just abilities) appear whispered when they should be.
Can you check if you perhaps have some module on your Foundry installation which might be affecting it somehow?

**Comment by Koppartikus** [comment]
Tried the following:

- Disabled all modules
- Disabled all modules except Beyond20 Companion Module
- Re-installed extension on Firefox
- Created a new test world in FoundryVTT and tested with no modules and Beyond20 Companion Module only

Results are unfortunately the same. Is there anything else I can attempt to test, or something I may have missed that may be causing the issue?

**Comment by kakaroto** [comment]
Argg, it drove me crazy. I just couldn't reproduce it, but @Aeristoka was able to reliably reproduce it. Eventually, somehow, for some reason, I changed this :
![image](https://user-images.githubusercontent.com/27990/151075771-9785dfa2-0c02-44a9-a49f-82227630644d.png)
From "Public Roll" to something else, then went back to "Public Roll" and now it does it consistently, and I cannot find any way to get it to reset back!
So, I started debugging and I dug deep into the bowels of Foundry to finally find that Foundry will override the whisper settings that we set if the type of message is a roll... 
I found an undocumented option in the code that we can pass to it to basically tell it not to override the whisper settings 🤦‍♂️ 
So.. finally, by looking at the logic of Foundry's code, I was able to figure out why it was working for me, and found a way to 'fix' it temporarily. Make a macro of type `script` and put this in it :
`game.settings.set("core", "rollMode", "roll")`
Then execute it...

It will basically change Foundry's dropdown setting next to the chat area from "Public roll" to.. well.. "Public roll", but this time, it won't override Beyond20's whisper setting... 
I'll do the proper fix in Beyond20 using the undocumented option and have it ready for next release.
Thanks for reporting the issue!
